webapp/src/context/AuthContext.tsx

```diff
@@ -60,177 +72,153 @@ const AppAuthProvider = ({ children }: { children: React.ReactNode }) => {
-        appSignOut();
-      }
-    });
-
   useEffect(() => {
-    let cancelled = false;
-
-    const bootstrap = async () => {
-      if (asg.isLoading) return;
-      if (localStorage.getItem("people-app-state") === "logout") {
-        setUI("logout");
-        return;
-      }
-
-      if (!localStorage.getItem("people-app-redirect-url")) {
-        localStorage.setItem(
-          "people-app-redirect-url",
-          window.location.href.replace(window.location.origin, "")
-        );
-      }
+    if (!localStorage.getItem(redirectUrl)) {
+      localStorage.setItem(
+        redirectUrl,
+        window.location.href.replace(window.location.origin, "")
+      );
+    }
+  }, []);
+
+  const setupAuthenticatedUser = async () => {
+    const [userInfo, idToken, decodedIdToken] = await Promise.all([
+      getBasicUserInfo(),
+      getIDToken(),
+      getDecodedIDToken(),
+    ]);
+
+    dispatch(
+      setUserAuthData({
+        userInfo: userInfo,
+        decodedIdToken: decodedIdToken,
+      })
+    );
+
+    new APIService(idToken, refreshToken);
+
+    await dispatch(getUserInfo());
+    await dispatch(loadPrivileges());
+    await dispatch(fetchAppConfig());
+  };
 
-      if (!asg.isAuthenticated) {
-        await signIn();
-        return;
-      }
+  useEffect(() => {
+    let mounted = true;
 
+    const initializeAuth = async () => {
       try {
-        const [basic, idToken, decoded] = await Promise.all([
-          getBasicUserInfo(),
-          getIDToken(),
-          getDecodedIDToken(),
-        ]);
-        if (cancelled) return;
-
-        dispatch(
-          setUserAuthData({ userInfo: basic, idToken, decodedIdToken: decoded })
-        );
-        new APIService(idToken, refreshToken);
+        setAppState("loading");
 
-        await dispatch(getUserInfo()).unwrap();
-        await dispatch(loadPrivileges()).unwrap();
+        if (state.isLoading) return;
 
-        if (!cancelled) setUI("active");
-      } catch (err: unknown) {
-        console.error("Failed to logging in :", err);
+        if (state.isAuthenticated) {
+          setAppState("authenticating");
+          await setupAuthenticatedUser();
 
-        const status = (err as any)?.response?.status ?? (err as any)?.status;
-        if (status === 401 || status === 403) {
-          if (!cancelled) {
-            localStorage.removeItem("people-app-state");
-            await signIn();
+          if (mounted) {
+            setAppState("authenticated");
           }
-          return;
-        }
-
-        try {
-          const stillAuth = await isAuthenticated();
-          if (!stillAuth) {
-            await signIn();
-            return;
+        } else {
+          const silentSignInSuccess = await trySignInSilently();
+
+          if (mounted) {
+            if (silentSignInSuccess) {
+              setAppState("authenticating");
+            } else {
+              setAppState("unauthenticated");
+            }
           }
```

Comment:
Check this with the [enum](https://github.com/wso2-open-operations/web-app-marketplace/pull/4/files#r2409439849),
```
if (mounted) {
  setAppState(silentSignInSuccess ? "authenticating" : "unauthenticated");
}
```